envoy/http/async_client.h

```diff
@@ -163,13 +170,52 @@ class AsyncClient {
      */
     virtual void reset() PURE;
 
+    /***
+     * Add destructor callback.
+     */
+    virtual void setDestructorCallback(StreamDestructorCallbacks callback) PURE;
+
+    /***
+     * Remove previously set destructor callback.
+     */
+    virtual void removeDestructorCallback() PURE;
+
+    /***
+     * Add watermark callbacks.
+     */
+    virtual void setWatermarkCallbacks(DecoderFilterWatermarkCallbacks& callbacks) PURE;
```

Comment:
ditto.  there's a lot of callbacks so maybe include when it'll be called and what should be done.

Reply:
Clarified semantics.